src/core/border_router/routing_manager.cpp

```diff
@@ -222,37 +223,18 @@ void RoutingManager::GenerateNat64Prefix(void)
 }
 #endif
 
-Error RoutingManager::LoadOrGenerateRandomOnLinkPrefix(void)
+void RoutingManager::GenerateOnLinkPrefix(void)
 {
-    Error error     = kErrorNone;
-    bool  generated = false;
-
-    if (Get<Settings>().Read<Settings::OnLinkPrefix>(mLocalOnLinkPrefix) != kErrorNone ||
```

Comment:
I think we can also deprecate this settings (maybe in a next PR)?

Reply:
Will do in a subsequent PR.